File: backend/app/ai_interview/schemas/interview.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class InterviewCreate(BaseModel):
    title: str
    department: Optional[str] = None
    experience_level: Optional[str] = "mid"
    skills: List[str] = []
    difficulty: str = "medium"
    duration: int = 30
    question_count: int = 5
    interview_type: str = "technical"

class InterviewOut(InterviewCreate):
    id: int
    status: str
    created_by: str
    created_at: datetime
    class Config:
        from_attributes = True

class QuestionOut(BaseModel):
    id: int
    interview_id: int
    question_text: str
    order: int
    class Config:
        from_attributes = True

File: frontend/src/features/ai-interview/types/interview.ts
export interface Interview {
  id: number;
  title: string;
  department?: string;
  experience_level?: string;
  skills: string[];
  difficulty: string;
  duration: number;
  question_count: number;
  interview_type: string;
  status: string;
  created_by: string;
  created_at: string;
}
export interface Question { id: number; interview_id: number; question_text: string; order: number; }
export interface Result {
  id: number; candidate_id: number; interview_id: number;
  technical_score: number; communication_score: number; confidence_score: number;
  problem_solving_score: number; final_score: number;
  strengths: string[]; weaknesses: string[]; suggestions: string[]; recommendation: string;
}
